src/test/lib/scheduler/operator_task_test.cpp

```diff
@@ -32,79 +39,138 @@ class OperatorTaskTest : public BaseTest {
 };
 
 TEST_F(OperatorTaskTest, BasicTasksFromOperatorTest) {
-  auto gt = std::make_shared<GetTable>("table_a");
+  const auto gt = std::make_shared<GetTable>("table_a");
   const auto& [tasks, root_operator_task] = OperatorTask::make_tasks_from_operator(gt);
 
   ASSERT_EQ(tasks.size(), 1);
-  root_operator_task->schedule();
+  EXPECT_NO_THROW(root_operator_task->schedule());
 
   EXPECT_TABLE_EQ_UNORDERED(_test_table_a, gt->get_output());
 }
 
 TEST_F(OperatorTaskTest, SingleDependencyTasksFromOperatorTest) {
-  auto gt = std::make_shared<GetTable>("table_a");
-  auto a = PQPColumnExpression::from_table(*_test_table_a, "a");
-  auto ts = std::make_shared<TableScan>(gt, equals_(a, 1234));
+  const auto gt = std::make_shared<GetTable>("table_a");
+  const auto a = PQPColumnExpression::from_table(*_test_table_a, "a");
+  const auto ts = std::make_shared<TableScan>(gt, equals_(a, 1234));
 
   const auto& [tasks, _] = OperatorTask::make_tasks_from_operator(ts);
-  for (auto& task : tasks) {
-    task->schedule();
-    // We don't have to wait here, because we are running the task tests without a scheduler
+  for (const auto& task : tasks) {
+    EXPECT_NO_THROW(task->schedule());
+    // We don't have to wait here, because we are running the task tests without a scheduler.
   }
 
-  auto expected_result = load_table("resources/test_data/tbl/int_float_filtered.tbl", ChunkOffset{2});
+  const auto expected_result = load_table("resources/test_data/tbl/int_float_filtered.tbl", ChunkOffset{2});
   EXPECT_TABLE_EQ_UNORDERED(expected_result, ts->get_output());
 }
 
 TEST_F(OperatorTaskTest, DoubleDependencyTasksFromOperatorTest) {
-  auto gt_a = std::make_shared<GetTable>("table_a");
-  auto gt_b = std::make_shared<GetTable>("table_b");
-  auto join = std::make_shared<JoinHash>(
+  const auto gt_a = std::make_shared<GetTable>("table_a");
+  const auto gt_b = std::make_shared<GetTable>("table_b");
+  const auto join = std::make_shared<JoinHash>(
       gt_a, gt_b, JoinMode::Inner,
       OperatorJoinPredicate{ColumnIDPair(ColumnID{0}, ColumnID{0}), PredicateCondition::Equals});
 
   const auto& [tasks, _] = OperatorTask::make_tasks_from_operator(join);
-  for (auto& task : tasks) {
-    task->schedule();
-    // We don't have to wait here, because we are running the task tests without a scheduler
+  for (const auto& task : tasks) {
+    EXPECT_NO_THROW(task->schedule());
```

Comment:
I don't get that. Then why not put a `expect_no_throw` around pretty  much everything?

Reply:
I confess that this is somehow a tricky question and `EXPECT_NO_THROW` does not always make sense. However, we also use it elsewhere, too. When we call something relying on other components, like in this case, it might be helpful to have some extra hint on what went wrong. Consider the following two test stubs:

```c++
TEST_F(TestFoo, Stub1) {
  const auto map = std::vector<uint16_t>{};
  map.at(0);
}

TEST_F(TestFoo, Stub2) {
  const auto map = std::vector<uint16_t>{};
  EXPECT_NO_THROW(map.at(0));
}
```

When executed, they result in a different granulyrity of error messages: 
```diff
+ [ RUN      ] TestFoo.Stub1
unknown file: Failure
C++ exception with description "vector" thrown in the test body.
- [  FAILED  ] TestFoo.Stub1 (0 ms)
+ [ RUN      ] TestFoo.Stub2
/Users/deyass/Documents/phd/hyrise/src/test/lib/utils/foo_test.cpp:295: Failure
Expected: map.at(0) doesn't throw an exception.
  Actual: it throws std::out_of_range with description "vector".
- [  FAILED  ] TestFoo.Stub2 (0 ms)
```
So it's easier to look at what actually went wrong. Would not do this everywhere, bit think here it's justified. However, I do not have a strong opinion and can remove as well.

Reply:
Hmm, okay.